File: go-service/models/purchase_order.go
package models

import "time"

// PurchaseOrder represents a purchase order to a supplier
type PurchaseOrder struct {
	ID                   int        `db:"Id" json:"id"`
	PONumber             string     `db:"PONumber" json:"poNumber"`
	SupplierID           int        `db:"SupplierId" json:"supplierId"`
	CreatedByUserID      int        `db:"CreatedByUserId" json:"createdByUserId"`
	ApprovedByUserID     *int       `db:"ApprovedByUserId" json:"approvedByUserId,omitempty"`
	TotalAmount          float64    `db:"TotalAmount" json:"totalAmount"`
	Status               string     `db:"Status" json:"status"`
	ExpectedDeliveryDate time.Time  `db:"ExpectedDeliveryDate" json:"expectedDeliveryDate"`
	DeliveredDate        *time.Time `db:"DeliveredDate" json:"deliveredDate,omitempty"`
	Notes                *string    `db:"Notes" json:"notes,omitempty"`
	CreatedAt            time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt            *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

// PurchaseOrderItem represents an item in a purchase order
type PurchaseOrderItem struct {
	ID              int       `db:"Id" json:"id"`
	PurchaseOrderID int       `db:"PurchaseOrderId" json:"purchaseOrderId"`
	ProductID       int       `db:"ProductId" json:"productId"`
	Quantity        int       `db:"Quantity" json:"quantity"`
	UnitPrice       float64   `db:"UnitPrice" json:"unitPrice"`
	TotalPrice      float64   `db:"TotalPrice" json:"totalPrice"`
	CreatedAt       time.Time `db:"CreatedAt" json:"createdAt"`
}

File: frontend/src/app/core/domain/purchase-order/purchase-order.ts
import { Supplier } from '../supplier/supplier';
import { User } from '../user/user';
import { OrderStatus } from '../../enums/order-status';

/**
 * Represents a Purchase Order Item
 */
export interface PurchaseOrderItem {
  /** ID */
  id: number;

  /** Purchase Order ID */
  purchaseOrderId: number;

  /** Product ID */
  productId: number;

  /** Quantity */
  quantity: number;

  /** Unit Price */
  unitPrice: number;

  /** Total Price */
  totalPrice: number;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}

/**
 * Represents a Purchase Order
 */
export interface PurchaseOrder {
  /** ID */
  id: number;

  /** PO Number - unique identifier */
  poNumber: string;

  /** Supplier ID - FK to Supplier */
  supplierId: number;

  /** Created By User ID - FK to User */
  createdByUserId: number;

  /** Approved By User ID - FK to User */
  approvedByUserId?: number | null;

  /** Total Amount */
  totalAmount: number;

  /** Status */
  status: OrderStatus;

  /** Expected Delivery Date */
  expectedDeliveryDate: Date;

  /** Delivered Date */
  deliveredDate?: Date | null;

  /** Notes */
  notes?: string | null;

  /** Supplier */
  supplier?: Supplier;

  /** Created By User */
  createdByUser?: User;

  /** Approved By User */
  approvedByUser?: User | null;

  /** Items - purchase order items */
  items?: PurchaseOrderItem[];

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}
